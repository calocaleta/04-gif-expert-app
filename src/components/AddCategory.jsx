import { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('One Punch');

    const onInputChanged = (event) => {
        setInputValue(event.target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        //console.log(inputValue);

        if(inputValue.trim().length <= 1 )  return;

        //onAddCategory(cat => [...cat,inputValue]);
        setInputValue('');
        onNewCategory(inputValue.trim());        
    }

    return (
        <form onSubmit={onSubmit}>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={ onInputChanged }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}