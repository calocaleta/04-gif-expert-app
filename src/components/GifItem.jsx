import PropTypes from 'prop-types'

export const GifItem = ({id,title,url}) => {
  return (
    <div className="card">
        <img src={url} key={id} alt={title} />
        <p>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}