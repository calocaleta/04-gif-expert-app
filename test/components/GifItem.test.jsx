import { render } from "@testing-library/react";
import {GifItem} from '../../src/components/GifItem'

describe('Pruebas en <GifItem />',()=>{
    
    const title = 'Saitama';
    const url = 'htto://onepunch.saitama.com';

    test('debe de hacer match con el snapshot', ()=> {
        const {container} = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

    });

});