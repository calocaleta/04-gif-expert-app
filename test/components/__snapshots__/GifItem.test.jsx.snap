// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="http://onepunch.saitama.com/index.php"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
